tests/functional/workloads/cnv/test_vm_snapshot_cloning_ops.py

```diff
@@ -290,3 +290,106 @@ def test_vm_snapshot_ops(
 
         if failed_vms:
             assert False, f"Test case failed for VMs: {', '.join(failed_vms)}"
+
+    def process_vm(
+        self,
+        vm_obj,
+        file_paths,
+        admin_client,
+        vm_clone_fixture,
+        vm_snapshot_restore_fixture,
+    ):
+        try:
+            source_csum = run_dd_io(vm_obj=vm_obj, file_path=file_paths[0], verify=True)
+            log.info(f"[{vm_obj.name}] Source checksum: {source_csum}")
+
+            log.info(f"[{vm_obj.name}] Creating clone of VM")
+            cloned_vm = vm_clone_fixture(vm_obj, admin_client)
+            run_dd_io(vm_obj=cloned_vm, file_path=file_paths[1])
+
+            log.info(f"[{vm_obj.name}] Creating snapshot of cloned VM")
+            restored_vm = vm_snapshot_restore_fixture(cloned_vm, admin_client)
+            restore_csum = cal_md5sum_vm(vm_obj=restored_vm, file_path=file_paths[0])
+
+            assert source_csum == restore_csum, (
+                f"[{vm_obj.name}] Failed: MD5 mismatch between source {vm_obj.name} "
+                f"and restored {restored_vm.name} cloned from '{vm_obj.name}'"
+            )
+
+            run_dd_io(vm_obj=restored_vm, file_path=file_paths[1])
+            log.info(f"[{vm_obj.name}] VM processing completed successfully.")
+        except Exception as e:
+            log.error(f"[{vm_obj.name}] Error during VM processing: {e}", exc_info=True)
+            raise
+
+    def process_all_vms_in_parallel(
```

Comment:
Was this function created to perform concurrent cloning and restore? If so, could you please rename the function accordingly and move it to a common location outside the test so it can be reused in other tests?